src/test/scala/uk/gov/hmrc/test/ui/pages/registration/RegistrationPage.scala

```diff
@@ -196,6 +196,35 @@ object RegistrationPage extends BasePage {
     this
   }
 
+  def provideEntityTypeDetailsForGeneralOrScottishPartnership(value: String): this.type = {
+    value match {
+      case "General partnership"  =>
+        SharedActions.selectLabelByPartialText("General partnership")
+        submitPage()
+        clickByCssSelector(cssForSaveAndContinue)
+        SharedActions
+          .enterDetails("A & J Company12")
+        submitPage()
+        onPage(BusinessSectorPage.heading)
+      case "Scottish partnership" =>
+        SharedActions.selectLabelByPartialText("Scottish partnership")
```

Comment:
Can we move this code into a shared function to remove the duplication, i.e. the code for the two partnership types is the same, it is just the values (data) that is different?